src/openforms/formio/dynamic_config/tests/test_referentielijsten_config.py

```diff
@@ -0,0 +1,372 @@
+        assert log.extra_data
+        self.assertEqual(
+            log.template, "logging/events/referentielijsten_failure_response.txt"
+        )
+        self.assertEqual(
+            log.extra_data["error"],
+            _("No results found from Referentielijsten API."),
+        )
+
+    @requests_mock.Mocker()
+    def test_request_exception(self, m):
+        m.get(
+            f"{self.service.api_root}items?tabel__code=tabel1",
+            exc=RequestException("something went wrong"),
+        )
+
+        with self.assertRaises(ServiceUnavailable):
+            SelectBoxes("selectboxes").mutate_config_dynamically(
+                self.component, self.submission, {}
+            )
+
+        self.assertNotIn("data", self.component)
+        log = TimelineLogProxy.objects.get(object_id=self.submission.form.id)
+        assert log.extra_data
+        self.assertEqual(
+            log.template, "logging/events/referentielijsten_failure_response.txt"
+        )
+        self.assertEqual(
+            log.extra_data["error"],
+            _(
+                "Exception occurred while fetching from Referentielijsten API: something went wrong."
+            ),
+        )
+
+
+@tag("gh-4993")
+class SubmissionStepDetailTest(SubmissionsMixin, APITestCase):
+    def test_get_submissionstep_detail_raises_error_for_referentielijsten_if_service_is_incorrect(
+        self,
+    ):
+        form = FormFactory.create(
+            generate_minimal_setup=True,
+            formstep__form_definition__configuration={
+                "components": [
+                    {
+                        "key": "selectboxes",
+                        "type": "selectboxes",
+                        "label": "Selectboxes",
+                        "values": [{"label": "", "value": ""}],
+                        "dataType": "string",
+                        "openForms": {
+                            "code": "tabel1",
+                            "dataSrc": "referentielijsten",
+                            "service": "non-existent",
+                            "translations": {},
+                        },
+                        "id": "ew0bwv7",
+                    }
+                ]
+            },
+        )
+        submission = SubmissionFactory.create(form=form)
+        endpoint = reverse(
+            "api:submission-steps-detail",
+            kwargs={
+                "submission_uuid": submission.uuid,
+                "step_uuid": form.formstep_set.get().uuid,
+            },
+        )
+        self._add_submission_to_session(submission)
+
+        response = self.client.get(endpoint)
+
+        self.assertEqual(response.status_code, status.HTTP_503_SERVICE_UNAVAILABLE)
+        # the detail message should be passed for errors that occur when fetching referentielijsten
+        self.assertEqual(
+            response.json()["detail"],
+            "Could not retrieve options from Referentielijsten API.",
+        )
```

Comment:
Can you either fix the language to English or mark the error message as translatable? This test will otherwise break when we add the Dutch translations.